### DISCUSSION TITLE
sort macro widget

### DISCUSSION THREAD

**Discussion by jmlat** [cncjs/cncjs#778] [Q&A]
Hello.
Is there a way to sort macros in the widget in order to have the one i use the most often above others ?

**Comment by MitchBradley** [comment]
One way is to use a text editor to edit the .cncrc file to place the macros in the order that you want.  The .cncrc file is located in the home directory on your computer, for example in C:\Users\<yourname>\.cncrc on Windows.  The macros section looks like this:
```
    "macros": [
        {
            "id": "c93bd9f8-5427-4703-9687-3483be4fbb20",
            "mtime": 1661544307125,
            "name": "Foo",
            "content": "G1 X10\n%wait\n"
        },
        {
            "id": "37b32ef8-f4a1-4429-8e8f-2f7c22aa2211",
            "mtime": 1661544301552,
            "name": "TorchOff",
            "content": "M5"
        }
    ],
```
You can just move the { } sections to whatever order you prefer.  Be careful that there is a comma between the {} sections, but there is no comma after the last }, otherwise cncjs will not be able to process the file correctly.

This is not the easy-to-use GUI technique that you were looking for, but it is what there is right now, without requiring a lot of coding that might not happen anytime soon.

**Comment by jmlat** [comment]
Thanks  MitchBradley
it seems to  be quite simple and it 'd be ok for me, but...
in C:\Users<yourname> I have not .cncrc file. ther is a .cncjs-sessionsdirectory with one file: UoUebyYzAR_UZOdcvgJw6gQK0JyOR84M.json
I tried to edit it , but it is noot like you described...

**Reply by MitchBradley** [reply]
Maybe the file is hidden and you have to do something to tell Windows to show hidden files.

**Reply by jmlat** [reply]
no I can see hidden files.

**Reply by jmlat** [reply]
Oh I got it... I was blind sorry !

**Reply by jmlat** [reply]
just perfect. thanks a lot
